File: Kotlin/src/main/kotlin/2920.kt
import java.util.Scanner

fun main() = with(Scanner(System.`in`)) {
    val n = mutableListOf<Int>()
    var ascending = 0
    var descending = 0
    for (i in 0 until 8) n.add(nextInt())
    for (i in 1..8) {
        if (n[i-1] == i) ascending++
        else if (n[8-i] == i) descending++
    }
    if (descending == 8) println("descending")
    else if (ascending == 8) println("ascending")
    else println("mixed")
}

File: C/2920.cpp
// 2920번: 음계
#include <iostream>

using namespace std;

int main() {
    int note = 0, cnt = 0;
    string scale;

    for (int i = 1; i <= 8; i++) {
        cin >> note;
        if (note == i)
            cnt++;
        else if (note == 9 - i)
            cnt--;
    }

    switch (cnt) {
        case 8: scale = "ascending"; break;
        case -8: scale = "descending"; break;
        default: scale = "mixed";
    }

    cout << scale;

    return 0;
}